elastic/kibana
# Expose anonymous access through a switch in sharing menu

## Summary

Closes https://github.com/elastic/kibana/issues/83650

See "Release note" below for more info.

### How to test

To test this feature, you should add the following to your `kibana.yml`:

```yml
xpack.security.authc.providers:
  basic.basic1:
    order: 0
  anonymous.anonymous1:
    order: 1
    credentials:
      username: "anonymous"
      password: "anonymous"
```

Then you will need to create a Kibana user with username `anonymous` and password `anonymous`. You can also create a dedicated Kibana role for the anonymous user, in that role you need to give "read" access to anonymous user for Dashboard, Discover and Visualize apps; and you need to give that user also access to Kibana Index Patterns. And you should be able to create URLs that can be accessed anonymously.

### Checklist

Delete any items that are not applicable to this PR.


### For maintainers


---

## Release note

[Anonymous authentication](https://www.elastic.co/guide/en/kibana/7.11/kibana-authentication.html#anonymous-authentication) feature is now exposed in Dashboard, Discover and Visualize apps in the sharing menu. If you enable an anonymous user with read access for Dashboard, Discover or Visualize apps you can create a sharing link which allows unauthenticated anonymous user to view those apps if you check "Public URL" switch in the sharing menu.

![image](https://user-images.githubusercontent.com/9773803/105970370-96b28500-6089-11eb-8b36-545657c1eb81.png)

![image](https://user-images.githubusercontent.com/9773803/105970424-a7fb9180-6089-11eb-8843-a45dc7fc29fc.png)

![image](https://user-images.githubusercontent.com/9773803/105970503-bcd82500-6089-11eb-9d7d-b2793cda8707.png)

![image](https://user-images.githubusercontent.com/9773803/105970588-d2e5e580-6089-11eb-8ece-0c6775842d8f.png)

![image](https://user-images.githubusercontent.com/9773803/105970653-e5f8b580-6089-11eb-8a9a-02f5dab7135b.png)
